fmutils.go

```diff
@@ -77,7 +77,20 @@ func (mask NestedMask) Filter(msg proto.Message) {
 				return true
 			}
 
-			if fd.IsList() {
+			if fd.IsMap() {
+				xmap := rft.Get(fd).Map()
+				xmap.Range(func(mk protoreflect.MapKey, mv protoreflect.Value) bool {
+					if _, ok := m[mk.String()]; !ok {
+						xmap.Clear(mk)
+					} else {
+						if mi, ok := m[mk.String()]; ok && len(mi) > 0 {
```

Comment:
Please, rewrite this part (and the similar part in the `Prune()` function below) so that there is only 1 lookup operation on the map:

```go
if v, ok := m[mk.String()]; ok && len(v) > 0 {
  v.Filter(mv.Message().Interface())
} else {
  xmap.Clear(mk)
}
```

Also, I think we need to check whether `mv` is actually a proto message before calling the `Filter()` function.

Reply:
Ah, of course.